### ISSUE TITLE
Instructions for how to update an app that used to use TimelineJS2

### ISSUE THREAD

**Issue by gotjoshua** [NUKnightLab/TimelineJS3#202]
Greetings,
I am starting a fork of timemapper, and I'd like to first get its submodule recline functioning with the new TimelineJS3. I could not find any migration information in the docs, did i miss something?

Can someone provide code examples and/or instructions for what must be changed to get updated?
many thanks!

**Comment by JoeGermuska** [comment]
You haven't missed anything, but you're the first developer we've heard from with an interest in migrating. TimelineJS has always had a vanishingly small percentage of users with an interest in the internals and details of the code.

We designed TimelineJS3 as a ground-up rewrite to get out of some inflexibilities that had developed in the old code base, so it's hard to even say if there's a migration path as such. 

We don't have the resources to develop a migration guide given that people who worked with the old code may have taken advantage of all kinds of things which were circumstantial choices or intended as private implementation details.

However, we're hoping that this new version is better designed for a developer community to work with and extend, and we're trying to create more clear and better organized [documentation](http://timeline3.knightlab.com/docs/index.html) toward that end. 

In short, you'll probably have to dive in and figure it out, but we'll do our best to help decode things. We also welcome pull requests for code or documentation. (Do check out [our contributors doc.](https://github.com/NUKnightLab/TimelineJS3/blob/master/CONTRIBUTING.md))

I'm going to close this issue, since we can't dedicate resources to the migration guide, but for now, this ticket is as good a place as any to ask questions if you need our help updating.

**Comment by gotjoshua** [comment]
Thanks Joe!
Well the first thing that I notice is that the old system refers to VMM and the new to VCO...
And then when I try to initiate a new VCO.timeline it seems to require the JSON object upfront (as a required parameter) whereas before it only got the div where it would be initialized into...

Is it a correct first step to replace VMM with VCO, and to be sure that I have the JSON beforehand so i can pass it in with the Object init?

**Comment by JoeGermuska** [comment]
Yes, when you make a Timeline instance, you need to either have the JSON or a URL to the JSON.

See http://timeline3.knightlab.com/docs/instantiate-a-timeline.html for hopefully more complete information. Help us improve it :)
